### ISSUE TITLE
Long compile time on 100K~ bytes static files

### ISSUE THREAD

**Issue by arowM** [cdepillabout/servant-static-th#1]
It takes long compile time when one of the front-end files becomes to be around 100K bytes.
A 100K bytes js files are likely such as one it imports `babel-polyfill`.

After installing dependent libraries,
`stack build --flag servant-static-th:buildexample` takes

* 19 seconds on current latest master (7ae9924)
* 350 seconds with [92K bytes js file](https://github.com/arowM/servant-static-th/commit/1d3bed480d7692e26584ef99b7d9c1cda9074956)

It must be a sort of bottleneck to let people to use this fantastic `servant-static-th` in real world applications.

**Comment by cdepillabout** [comment]
I've created a temporary fix for this in #2.

It would be nice to figure out a good way to include files with arbitrary bytes in a Haskell source file.  Is it reasonable to store them in a `String`?  What's the best way to convert the `String` back and forth to a `ByteString`?  Something like `Data.ByteString.Char8.pack` and `Data.ByteString.Char8.unpack`?  Or maybe something like the following that doesn't make any assumptions about the current locale:

```haskell
import Control.Monad.IO.Class (MonadIO(liftIO))
import Data.ByteString (ByteString)
import Data.ByteString.Unsafe (unsafePackMallocCStringLen)
import Foreign.C (newCStringLen)

-- | Convert a 'String' to a 'ByteString' with the encoding for the current
-- locale.
--
-- >>> convertStringToRawByteString "hello"
-- "hello"
convertStringToRawByteString :: MonadIO m => String -> m ByteString
convertStringToRawByteString str = liftIO $ do
  cStringLen <- newCStringLen str
  unsafePackMallocCStringLen cStringLen
{-# INLINABLE convertStringToRawByteString #-}
```

**Comment by cdepillabout** [comment]
This should be fixed in #3.

**Comment by arowM** [comment]
Thanks!
